.github/workflows/build-frontend.yml

```diff
@@ -13,29 +13,58 @@ on:
       - "!.github/workflows/stellar-dominion-service*.yml"
 
 jobs:
+  # https://github.com/dorny/paths-filter
+  analyze-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      user-dashboard: ${{ steps.filter.outputs.user-dashboard }}
+      stellar-dominion: ${{ steps.filter.outputs.stellar-dominion }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          base: ${{ github.ref_name }}
```

Comment:
We need to set this to the current branch: otherwise the default branch is taken (in our case `master`). This means that for a given feature branch, as soon as there's a change of a file in the pathes related to a frontend service compared to **master** then it will always build the docker image for this service.

This is not exactly what we want: we would like the build to happen in case there were some changes compared to the **previous commit**.